libraries/rib-base/src/main/java/com/badoo/ribs/clienthelper/childawareness/ChildAwareRegistryImpl.kt

```diff
@@ -0,0 +1,71 @@
+package com.badoo.ribs.clienthelper.childawareness
+
+import androidx.lifecycle.Lifecycle
+import com.badoo.ribs.core.Node
+import com.badoo.ribs.core.Rib
+import com.badoo.ribs.core.lifecycle.CombinedLifecycle
+import kotlin.reflect.KClass
+
+class ChildAwareRegistryImpl : ChildAwareRegistry {
+
+    private val registeredCallbacks: MutableList<CallbackInfo<*, *>> = ArrayList()
+    override lateinit var node: Node<*>
+        private set
+
+    override fun init(node: Node<*>) {
+        this.node = node
+    }
+
+    override fun <T : Rib, R : Rib> whenChildrenAttached(
+        lifecycle: Lifecycle,
+        child1: KClass<T>,
+        child2: KClass<R>,
+        callback: ChildrenAvailableCallback<T, R>,
+    ) {
+        val info = CallbackInfo(child1, child2, callback, lifecycle)
+        registeredCallbacks += info
+        // TODO CHECK EXISTING CHILDREN
+    }
+
+    override fun onChildBuilt(child: Node<*>) {
+        // onChildBuilt is invoked before this child is added to node.children list
+        for (info in registeredCallbacks) {
+            val second = info.getOther(child::class) ?: continue
+            val connectToList = node.children.filter { it::class.isInstance(second) }
```

Comment:
May be broken for the case of portal. Bc we are not adding child shown in portal to the list of children in the virtual parent